Tests/Marketplace/search_and_install_packs.py

```diff
@@ -413,381 +600,348 @@ def install_packs(client: demisto_client,
+                )
 
-    :param client: Demisto-py client to connect to the server.
-    :param host: FQDN of the server.
-    :param service_account: The full path to the service account json.
-    :return: None. Prints the response from the server in the build.
-    """
-    all_packs = []
+            elif isinstance(ex, HTTPError | HTTPWarning):
+                logging.warning(f"Failed to perform HTTP request : {ex}")
 
-    # Initiate the GCS client and get the production bucket
-    storage_client = init_storage_client(service_account)
-    production_bucket = storage_client.bucket(GCPConfig.PRODUCTION_BUCKET)
-    logging.debug(f"Installing all content packs for nightly flow in server {host}")
+            else:
+                logging.warning(
+                    f"API call to fetch dependencies of '{pack_id}' has failed.\nError: {ex}."
+                )
+                logging.debug(f"Stack trace:\n{traceback.format_exc()}")
 
-    # Add deprecated packs to IGNORED_FILES list:
-    for pack_id in os.listdir(PACKS_FULL_PATH):
-        if is_pack_deprecated(pack_id=pack_id, production_bucket=False):
-            logging.debug(f'Skipping installation of hidden pack "{pack_id}"')
-            IGNORED_FILES.append(pack_id)
+            if attempt:  # There are remaining retry attempts
+                logging.debug(f"Sleeping for {sleep_interval} seconds and retrying...")
+                time.sleep(sleep_interval)
 
-    for pack_id in os.listdir(PACKS_FULL_PATH):
-        if pack_id not in IGNORED_FILES:
-            pack_version = get_latest_version_from_bucket(pack_id, production_bucket)
-            if pack_version:
-                all_packs.append(get_pack_installation_request_data(pack_id, pack_version))
-    install_packs(client, host, all_packs)
+            else:  # No retry attempts remaining
+                logging.error(f"All {attempts_count} attempts have failed.")
+                break
 
+    return None
 
-def install_all_content_packs_from_build_bucket(client: demisto_client, host: str, server_version: str,
-                                                bucket_packs_root_path: str, service_account: str,
-                                                extract_destination_path: str):
-    """ Iterates over the packs currently located in the Build bucket. Wrapper for install_packs.
-    Retrieving the metadata of the latest version of each pack from the index.zip of the build bucket.
 
-    :param client: Demisto-py client to connect to the server.
-    :param host: FQDN of the server.
-    :param server_version: The version of the server the packs are installed on.
-    :param bucket_packs_root_path: The prefix to the root of packs in the bucket
-    :param service_account: Google Service Account
-    :param extract_destination_path: the full path of extract folder for the index.
-    :return: None. Prints the response from the server in the build.
+def search_pack_and_its_dependencies(
```

Comment:
Maybe a more suitable name would be: get_pack_and_its_dependencies?
Your call.

Reply:
fixed